changelogs/unreleased/3830-detect-string-interpolation-in-dict-keys.yml

```diff
@@ -0,0 +1,6 @@
+description: An exception is now raised whenever a curly bracket is used in a string used as a key in a dictionary
+change-type: patch
+destination-branches: [master, iso4, iso5]
```

Comment:
Strictly speaking: only master

We could get away with iso5 now, but let's keep it at master.

Reply:
> @Hugo-Inmanta are you already familiar with the different branches and their meaning/contract?

In the general sense yes. As I understand it ISO5 is the currently supported version, master is the soon-to-come ISO6.
Now that we discuss it it makes sense not to push that change to older versions and only to master

Reply:
That's right, but the main deciding factor for which branches to merge into is the following:
- iso4 is in maintenance mode (also referred to as EOS (end of sales): we need to keep supporting it but we don't provide new features. It should receive only SemVer patches.
- iso5 is our main iso release: we add new features here but since it's already being used we mustn't add breaking changes. SemVer patch/minor.
- master is our future iso release: it isn't used anywhere outside of the open source product and our internal stuff. Here we have full freedom.

Since this is technically a breaking change it should only go into master. An argument could be made as well to make this a `major` change rather than a `patch` for the same reasons. I have no strong opinion on that last part.

Reply:
Is there a list for the type changes ? I guess `minor`, `major` and `patch` ?

Reply:
Indeed. You can read more about SemVer [here](https://semver.org/) if you're interested.

Reply:
We use these semantics for all our versioning except for the `inmanta` (open source) product where we use `<year>.<minor>` for legacy reasons.